Task4/SingleltonPattern.java

```diff
@@ -0,0 +1,42 @@
+package Task4;
+
+class Singleton {
+	private static Singleton singleObj = null;
+
+	// making the constructor private to prevent obj creation from outside
+	private Singleton() {
+
+	}
+
+	// create obj will let only one obj creation
+	synchronized static Singleton createObj() {
```

Comment:
Instead of having `createObj()` as the method name, it would be better to have it as `getInstance()` as the user won't know whether a new object is created or an existing instance is returned.